.github/workflows/build-test.yaml

```
@@ -1,16 +1,18 @@
 name: Build, test
 
 on:
-  schedule:
-    - cron: "0 2 * * 1,3,5"  # 02:00 UTC every Mon,Wed,Fri
+  push:
+    branches:
+      - main
+  pull_request:
```

What do you think of limiting the GitHub Actions to only run when changes to the ipynb files are made? Since they are the target of the tests.

```suggestion
on:
  push:
    branches: main
    paths: '*.ipynb'
  pull_request:
    branches: main
    paths: '*.ipynb'
```

---

Good point. What about limiting actions to ipynb files on push only and having no limits for PRs?

---

Not sure if I see the advantage/difference.  Can you please elaborate?

I had allegedly added push events so that when the `main` branch gets a merge commit it can get tested, so not sure if what you propose would allow to test the `main` branch when the requirements or Python versions change.

Note that I had first added `on: [push, pull_request]` and that triggered duplicate builds on this PR.

---

> Note that I had first added on: [push, pull_request] and that triggered duplicate builds on this PR.

I suspect if you want to have both `push` and `pull_request` event triggers, you will also need to specify the type of PR event that triggers the workflow. I think pushing should also trigger a pull request action via `synchronize` (if branch is linked to an open PR) that is causing the duplicate build. 

As for limiting to ipynb files on push only to the `main` branch, I think I also am not seeing the full picture here.

---

Disregard that. I think going forward, it may be better to use the `[skip ci]` commit message feature for commits that only affect the carpentries template. Does this sound like a better option rather than the earlier suggestion of triggering only when ipynb files change?

---

> > Note that I had first added on: [push, pull_request] and that triggered duplicate builds on this PR.
> 
> I suspect if you want to have both `push` and `pull_request` event triggers, you will also need to specify the type of PR event that triggers the workflow. I think pushing should also trigger a pull request action via `synchronize` (if branch is linked to an open PR) that is causing the duplicate build.

I'm not getting it. Please go ahead and amend the commit if you see a clear way for that.

> Disregard that. I think going forward, it may be better to use the [skip ci] commit message feature for commits that only affect the carpentries template. Does this sound like a better option rather than the earlier suggestion of triggering only when ipynb files change?

I am not convinced about it; the website check CI should run if changes to the episode markdowns are made provided that the check ensures that they comply with the style and the site builds.

---

> I am not convinced about it; the website check CI should run if changes to the episode markdowns are made provided that the check ensures that they comply with the style and the site builds.

I think this is the only workflow that makes use of the `[skip ci]` flag. The other 2 workflows would still run despite using the flag.

---

> I think this is the only workflow that makes use of the [skip ci] flag. The other 2 workflows would still run despite using the flag.

:+1: You're right.